### DISCUSSION TITLE
ESP32C3 with Tasmota + CC11010 (868Mhz) + WM-BUS

### DISCUSSION THREAD

**Discussion by radek-turecki** [arendst/Tasmota#22520] [General]
I would like to kindly ask if perhaps any of you have already done a solution on ESP32 with Tasmota in combination with CC1101 to read the electricity meter over WM-BUS?

Do you have any experience that you could share? I would appreciate your help.

**Comment by radek-turecki** [comment]
Hello again,

I connected CC1101 to Lutaos ESP32C3 like on the picture:
![image](https://github.com/user-attachments/assets/4256c819-7094-435a-91ba-3513eca5fd72)

and I prepared template:
`{"NAME":"ESP32C3-CC1101","GPIO":[4544,4576,736,704,0,0,0,768,0,0,672,0,0,0,0,0,0,0,0,0,0,0],"FLAG":0,"BASE":1}`

referring to Tasmota doc it means:
`{"NAME":"ESP32C3-CC1101","GPIO":[CC1101 GDO0-CC1101 RX Pin,CC1101 GDO2-CC1101 TX Pin,SPI CLK1-Hardware SPI,SPI MOSI1-Hardware SPI,0,0,0,SPI CS1-Hardware SPI,0,0,SPI MISO1-Hardware SPI,0,0,0,0,0,0,0,0,0,0,0],"FLAG":0,"BASE":1}`

but on the form of module some values are blank (not filled) like on the picture:
![image](https://github.com/user-attachments/assets/b2bbd8c7-5a67-4c6a-92cc-0645183550df)

MQTT is set of course.

So the questions are:
1. Why values in the template are blank? Is there some error of Tasmota?
2. No any data from CC1101 is transmited in the Console (only information about settings like SSID, time, etc., but nothing more)
3. How can I activate CC1101 module? Hot to send any command from Tastmota to GPIOx?

Somebody can help me?

**Comment by arendst** [comment]
Tasmota doesn't have native CC1101 support. Your selected GPIO's belong to the Keeloq driver using CC1101.

In order to use these GPIO's you'll have to compile your own version of Tasmota uisng define USE_KEELOQ.

But as said, this driver only supports specific Keeloq shutters.

**Reply by radek-turecki** [reply]
Thank you @arendst for information. Unfortunately, I'm too weak (I don't know it :-()) to be able to compile my own version of Tasmot.

**Comment by nikito7** [comment]
Adapt this to tasmota. Or original wmbusmeters. 

https://github.com/SzczepanLeon/esphome-components/tree/main/components/wmbus

It's need to interface tasmota driver system with wmbusmeters code.

**Reply by radek-turecki** [reply]
Yes from what I researched the topic, it looks like this solution is based on :
[(https://github.com/wmbusmeters/wmbusmeters/)]

and is the most advanced and developed and supports a very large number of electricity, water and gas meters.

Unfortunately, I am not a programmer to use this code myself :-(

Maybe Tasmot developers will consider implementing CC1101 in future versions of Tasmot? It would be great if Tasmota supports WM-BUS.

**Reply by sfromis** [reply]
With open source projects like Tasmota, most such device additions are created by someone outside of the "core" group, having the interest and time to create a driver to submit to the project.
